pyquil/api/qvm.py

```diff
@@ -0,0 +1,305 @@
+##############################################################################
+# Copyright 2016-2017 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+
+import json
+
+from six import integer_types
+
+from pyquil.quil import Program
+from pyquil.wavefunction import Wavefunction
+from ._base_connection import BaseConnection, validate_noise_probabilities, validate_run_items, TYPE_MULTISHOT, \
+    TYPE_MULTISHOT_MEASURE, TYPE_WAVEFUNCTION, TYPE_EXPECTATION, get_job_id
+
+
+class QVMConnection(BaseConnection):
+    """
+    Represents a connection to the QVM.
+    """
+
+    def __init__(self, endpoint=None, api_key=None, user_id=None, skip_queue=True,
+                 gate_noise=None, measurement_noise=None, random_seed=None):
+        """
+        Constructor for QVMConnection. Sets up any necessary security, and establishes the noise
+        model to use.
+
+        :param endpoint: The endpoint of the server (default behavior is to read from config file)
+        :param api_key: The key to the Forest API Gateway (default behavior is to read from config file)
+        :param user_id: Your userid for Forest (default behavior is to read from config file)
+        :param skip_queue: Enabling this parameter may improve performance for small, quick programs.
+                           To support larger programs, set it to False.
+        :param gate_noise: A list of three numbers [Px, Py, Pz] indicating the probability of an X,
+                           Y, or Z gate getting applied to each qubit after a gate application or
+                           reset. (default None)
+        :param measurement_noise: A list of three numbers [Px, Py, Pz] indicating the probability of
+                                  an X, Y, or Z gate getting applied before a a measurement.
+                                  (default None)
+        :param random_seed: A seed for the QVM's random number generators. Either None (for an
+                            automatically generated seed) or a non-negative integer.
+        """
+        self.skip_queue = skip_queue
+        if not endpoint:
+            if skip_queue:
+                endpoint = 'https://api.rigetti.com'
+            else:
+                endpoint = 'https://job.rigetti.com/beta'
+
+        super(QVMConnection, self).__init__(endpoint=endpoint, api_key=api_key, user_id=user_id)
```

Comment:
this means pyquil_config will overwrite by skip_queue param (I think), which is not what I'd want.